### DISCUSSION TITLE
Unable to access file contents

### DISCUSSION THREAD

**Discussion by MaybeThisIsRu** [vercel/vercel#5014] [Help] [answered]
I use Eleventy and receive guest comments as files to my repository [using an Express API hosted elsewhere](https://github.com/hirusi/web-comments).

The following code runs fine on my local environment, but complains on Vercel that it could not find the `cache/comments/` directory and creates it.

```js
const fs = require("fs");
const path = require("path");

const COMMENTS_DIR = path.join(process.cwd(), "/cache/comments");
const COMMENTS = [];

const getCommentFilesList = () =>
	new Promise((resolve, reject) => {
		if (!fs.existsSync(COMMENTS_DIR)) {
			console.log(">>> Comments dir doesn't exist, creating...");
			fs.mkdirSync(COMMENTS_DIR, {
				rescursive: true
			});
		}
		fs.readdir(COMMENTS_DIR, null, (error, files) => {
			if (error) reject(error);
			resolve(files);
		});
	});
```

```
13:39:05.944 | >>> Collecting comments from /vercel/76f3ac9f/cache/comments
13:39:05.945 | >>> Comments dir doesn't exist, creating...
```

This is unexpected as the directory does exist in the repo, and is not gitignored either. What am I doing wrong here?

The [files ignored by Vercel](https://vercel.com/docs/build-step#ignored-files-and-folders) do not include `cache` directory.

**Comment by styfle** [comment, marked as answer]
Try changing `process.cwd()` to `__dirname` instead and use the relative path to the directory.

**Reply by MaybeThisIsRu** [reply]
That did not work unfortunately, still the same output on production but still works fine on local with this change.

```
09:52:43.937 | >>> Collecting comments from /vercel/acf1aad/cache/comments
09:52:43.937 | >>> Comments dir doesn't exist, creating...
09:52:43.938 | >>> Comment files list received
09:52:43.938 | >>> Error ocurred while getting all comments: Error: No comment files on disk
```

**Reply by styfle** [reply]
Is the `./cache/comments` checked into git? You might try doing `touch ./cache/comments/.gitkeep` and commit.

That being said, your comments will disappear each time you deploy if you're planning to write to that directory.

I suggest storing comments in a database instead so it will persist between deploys.

**Reply by MaybeThisIsRu** [reply]
Yep, they're included in the repo. I checked the git tree through Vercel and they sure are.

So these are HTML pages, and the comments get included at build time. They're not going to disappear between deploys. Maybe there's another issue here - I do have a couple of other JSON caches working fine.

**Reply by styfle** [reply]
Is the repository open source so I can take a look?

If not, can you create an example repo to reproduce this issue?

**Reply by MaybeThisIsRu** [reply]
It's a private repo though I can add you as a contributor temporarily?

This seems to work on Forestry CMS' preview servers as well, so I think it really might be something with the way Vercel builds it.

**Reply by styfle** [reply]
Sure 👍

**Reply by MaybeThisIsRu** [reply]
`__dirname` did the trick. I had one `..` input incorrectly. `process.cwd()` doesn't seem to work.
